woocommerce/woocommerce
# Use SVG and Dashicons instead of custom font for admin bar icons

### All Submissions:






### Changes proposed in this Pull Request:



This PR changes the way the that icons for admin menu items are provided, from CSS with custom fonts, to SVG and Dashicons. This is to allow the icons to be displayed in Calypso when Nav Unification is enabled. See pbIJXs-zH-p2 for more information on Nav Unification and WooCommerce.

The icons for WooCommerce, Products, and Reports have been updated in both wp-admin and Calypso.

_Note: I am not familiar with how to get the Reports top-level menu item to appear... if anybody can let me know/test this, it would be much appreciated._

_Note: SVG icons in Calypso currently appear with incorrect color (see https://github.com/Automattic/wp-calypso/issues/49145). This is expected to be addressed separately and generally for Nav Unification._

_Note: The position of the Products menu item in Calypso is incorrect and will be addressed separately in https://github.com/Automattic/jetpack/pull/18758._

#### Before and After in wp-admin

![Before and After wp-admin](https://user-images.githubusercontent.com/2098816/107791966-93262b80-6d22-11eb-8225-7ef3dce3ce16.png)

#### Before and After in Calypso

![Before and After Calypso](https://user-images.githubusercontent.com/2098816/107791979-96b9b280-6d22-11eb-8c9c-e6fafea15f45.png)

### How to test the changes in this Pull Request:

#### General wp-admin testing without Nav Unification

1. Checkout this branch.
2. Access wp-admin.
3. Verify that the WooCommerce and Products icons appear the same as before.

#### Testing Nav Unification

Make your test site act like an "Atomic" site:

4. Connect Jetpack on your test site to WordPress.com.
5. Clone `wpcomsh` repo (see https://github.com/Automattic/wpcomsh for details).
6. Checkout the `add/admin-menu-request-impersonation` branch (577-gh-Automattic/wpcomsh)
7. Add a plugin to `mu-plugins` with the following line, to enable Nav Unification:

```
add_filter( 'jetpack_load_admin_menu_class', '__return_true' );
```

8. Access your connected site from WordPress.com (Calypso) and verify that the menu items appear correctly (excluding issues described above).

### Other information:




### Changelog entry

> Dev - Updated admin bar icons to use SVG and Dashicons instead of custom font.
